add collisionsprite and introduce mask threshold

lily-seabreeze opened the pull request.

lillian-gardenia-seabreeze commented on sappho/animatedsprite.py:
```diff
@@ -8,12 +8,16 @@
 from PIL import Image
 
 
+# TODO: add mask, which then animatedsprite points to
```
lol outdated comment

lily-seabreeze commented:
collision closer to 100% working
Rename "physicalsprite" stuff to "collisionsprite" because
that encompasses its scope more accurately.

Update `demo.py` to use `collisionsprite` and do collision
accurately. I may need to cut some of the collision logic
therein to be put as method(s) fer `CollisionSprite`.

There remains a bug whereas the player is capable of
becoming ensnared once collision has been detected.

Removed the bitbucket install of pygame from `requirements/main.txt`
because it works only on a meager handful of platforms. I'll have to
document pygame installation on various platforms. Maybe that
Hypatia Engine bootstrapper is actually a good thing. We really
should pair that with Jenkins testing all OSs we support in the
bootstrapper.

Changed tilesheet rules to use new flag term "solid" which
replaces solid_block.

Add Tile collision support with automasking and rect creation
when `solid` is specified.

Implement `Tile.copy()` so tiles from a tilesheet may be copied
and be capable of differences on the tile map, such as rect
coordinate.

Remove `get_solid_blocks()` in `TileMap` because it's replaced
by `set_solid_toplefts()` (see docstring for details).
